### PR TITLE
Update complement.sh to match new public API shape

### PR DESCRIPTION
**Pull request by kegsay**
Sister PR to https://github.com/matrix-org/complement/pull/666

Context: https://github.com/matrix-org/complement/issues/654#issuecomment-1746613495

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [ ] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by DMRobertson** [comment]
Try adding the [complement.sh](http://complement.sh/) script in https://github.com/matrix-org/synapse/blob/a6abee36bc53feb4e5ffc6284af0b059f8f0ab2c/.github/workflows/tests.yml#L50-L58 to make sure CI actually runs here. (C.f. https://github.com/matrix-org/synapse/pull/14745 and friends)

**Comment by kegsay** [comment]
It runs now, and fails the same failing tests on `develop`, which means it is in fact running those MSCs.

File: scripts-dev/complement.sh [hunk 1]
```diff
@@ -214,7 +214,7 @@ fi
 
 extra_test_args=()
 
-test_tags="synapse_blacklist,msc3874,msc3890,msc3391,msc3930,faster_joins"
+test_packages="./tests/csapi ./tests ./tests/msc3874 ./tests/msc3890 ./tests/msc3391 ./tests/msc3930 ./tests/msc3902"
```
Comment (DMRobertson) [inline scripts-dev/complement.sh:217]: slightly unfortunate that you have to say `./tests` here all the time... but I guess this lets you specify test packages in some other repo too?

File: .github/workflows/tests.yml
Comment (DMRobertson) [inline .github/workflows/tests.yml:1]: [GHA says](https://github.com/matrix-org/synapse/actions/runs/6484314796)

> complement (monolith, Postgres)
Restore cache failed: Dependencies file is not found in /home/runner/work/synapse/synapse. Supported file pattern: go.sum

Not a problem and maybe not even introduced here. Don't suppose you can spot a quick fix in passing?

**Review by DMRobertson — APPROVED** [review]
Otherwise LGTM, thanks! I'll let you merge in tandem with the complement and Dendrite change.

File: scripts-dev/complement.sh (hunk 1 above)
Comment (kegsay) [reply scripts-dev/complement.sh:217]: Yes

File: .github/workflows/tests.yml
Comment (kegsay) [reply .github/workflows/tests.yml:1]: Didn't see anything obvious sadly.

**Pull request merged**